### PR TITLE
Common.DocumentationRef

### PR DESCRIPTION
**Pull request by HeikoTheissen**

### CODE REVIEW

File: vocabularies/Common.xml [hunk 1]
```diff
@@ -74,6 +74,17 @@
         <Annotation Term="Core.IsLanguageDependent" />
       </Term>
 
+      <Term Name="DocumentationRef" Type="Edm.String" Nullable="false" AppliesTo="EntityType Property Parameter ReturnType">
+        <Annotation Term="Core.IsURL" />
+        <Annotation Term="Core.Description" String="A URI referencing language-dependent documentation for the annotated model element" />
+        <Annotation Term="Core.Example">
+          <Record Type="Core.PrimitiveExampleValue">
+            <PropertyValue Property="Description" String="URN scheme understood by SAP Companion" />
+            <PropertyValue Property="Value" String="urn:sap:companion:DataElement:CURRENCY:G1Y_000" />
```
Comment (HeikoTheissen) [inline vocabularies/Common.xml:83]: Please update this example

**Review by uhlmannm — DISMISSED** [review]

File: vocabularies/Common.json [hunk 2]
```diff
@@ -59,6 +59,17 @@
       "@Core.Description": "A short, human-readable text suitable for tool tips in UIs",
       "@Core.IsLanguageDependent": true
     },
+    "DocumentationRef": {
+      "$Kind": "Term",
+      "$AppliesTo": ["EntityType", "Property", "Parameter", "ReturnType"],
```
[the remark this answers is not in the document]
Comment (uhlmannm) [reply vocabularies/Common.json:64]: Function, Action and FunctionImport are not relevant. ComplexType could be used. TypeDefinition also seems unneccessary.

File: vocabularies/Common.json [hunk 3]
```diff
@@ -59,6 +59,17 @@
...
+      "@Core.IsURL": true,
+      "@Core.Description": "A URI referencing language-dependent documentation for the annotated model element",
```
[the remark this answers is not in the document]
Comment (uhlmannm) [reply vocabularies/Common.json:67]: Which means that you could tag anything with the DocumentationRef? The UI would definitely only evaluate it at a few places. And GW would not send it everywhere. But from an annotation point of view that could still be okay. Just external viewers could expect too much.

[the remark this answers is not in the document]
Comment (ralfhandl) [reply vocabularies/Common.json:67]: I would not restrict it to what ABAP/GW/... currently supports as this is in Common and not in an ABAP vocabulary.

File: vocabularies/Common.xml [hunk 4]
```diff
@@ -74,6 +74,18 @@
         <Annotation Term="Core.IsLanguageDependent" />
       </Term>
 
+      <Term Name="DocumentationRef" Type="Edm.String" Nullable="false" AppliesTo="EntityType Property Parameter ReturnType">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.IsURL" />
+        <Annotation Term="Core.Description" String="A URI referencing language-dependent documentation for the annotated model element" />
+        <Annotation Term="Core.Example">
+          <Record Type="Core.PrimitiveExampleValue">
+            <PropertyValue Property="Description" String="URN scheme understood by SAP Companion" />
```
Comment (BerSie) [inline vocabularies/Common.xml:83]: This product has in the last two or three years at least three product name changes - not sure if I would mention it here. Is there a difference to "normal" URLs?

Comment (uhlmannm) [reply vocabularies/Common.xml:83]: Yes. This is an example from one of our test services.
> `<Annotation Term="SAP__common.DocumentationRef" String="urn:sap-com:documentation:key?=type=DE&id=SPRAS"/>`

So it is in the end pretty ABAP specific. If we would want to enhance it for other backends, also Companion would need to learn about that.

Comment (BerSie) [reply vocabularies/Common.xml:83]: > Yes. This is an example from one of our test services.
> 
> > `<Annotation Term="SAP__common.DocumentationRef" String="urn:sap-com:documentation:key?=type=DE&id=SPRAS"/>`
> 
> So it is in the end pretty ABAP specific. If we would want to enhance it for other backends, also Companion would need to learn about that.

OK - thanks, @uhlmannm . The question is if this urn schema was also used for xray, for FLP In-App Help, and Web Assistant, which is now used for SAP Companion - so either it's only relevant for this specific tool, or it's relevant for the in-app help framework from SAP. Usually we want to define here a "timeless" abstraction ...

Comment (HeikoTheissen) [reply vocabularies/Common.xml:83]: This is just an _example_.

Comment (BerSie) [reply vocabularies/Common.xml:83]: Thanks, @HeikoTheissen!
Nevertheless, it's only yesterday that I heard that there is the next name under way. I assume that in two years nobody knows what companion is/was. But don't mind - it was just a comment. For me the term itself is completely fine.

**Review by BerSie — DISMISSED** [review]

File: vocabularies/Common.xml (hunk 4 above)
Comment (uhlmannm) [reply vocabularies/Common.xml:83]: But in the end they evaluate the schema. Should we write rather "URN scheme containing the information to identify the help text"?

**Review by uhlmannm — DISMISSED** [review]

File: vocabularies/Common.xml [hunk 5]
```diff
@@ -74,6 +74,21 @@
         <Annotation Term="Core.IsLanguageDependent" />
       </Term>
 
+      <Term Name="DocumentationRef" Type="Edm.String" Nullable="false">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.IsURL" />
```
Comment (ralfhandl) [inline vocabularies/Common.xml:79]:
```suggestion
```

**Review by ralfhandl — CHANGES_REQUESTED** [review]
Is not a URL

**Review by ralfhandl — APPROVED** [review]

**Review by BerSie — APPROVED** [review]
under the given circumstances it's OK for me, but I will never understand why we make nearly everything in our system accessible via (OData) URL, but our documentation (static texts) can't be directly addressed by URLs.

**Pull request merged**